java/src/main/java/io/cucumber/gherkin/StringUtils.java

```diff
@@ -104,6 +104,15 @@ static boolean containsWhitespace(String input, int fromIndex, int toIndex) {
         return false;
     }
 
+    static String substringAndTrim(String input, int beginIndex) {
```

Comment:
The new method is not very efficient, as seen in the profiler. Checking whether the character is a space or not is about 50% of the method duration, and the `text` length pre-computed in `GherkinLine` constructor is not reused:
<img width="3506" height="189" alt="image" src="https://github.com/user-attachments/assets/7b362d80-b8c9-412d-b3e0-0b38e0016796" />
To be managed later...

Reply:
Interesting. `StringUtils.isWhiteSpace` only takes 16% of `StringUtils.substringTrimmed` for me. 

<img width="1774" height="228" alt="image" src="https://github.com/user-attachments/assets/e08209dd-5f57-4d00-96e7-967b49a38e97" />

>  and the text length pre-computed in GherkinLine constructor is not reused:

Ah good point. Didn't realize there would be call to `String.coder()` for every `length()` call.

Reply:
https://github.com/cucumber/gherkin/pull/444